pkg/clients/rds/rds.go

```diff
@@ -44,12 +45,17 @@ type Client interface {
 }
 
 // NewClient creates new RDS RDSClient with provided AWS Configurations/Credentials
-func NewClient(credentials []byte, region string) (Client, error) {
-	cfg, err := awsclients.LoadConfig(credentials, awsclients.DefaultSection, region)
-	if err != nil {
-		return nil, err
+func NewClient(ctx context.Context, credentials []byte, region string, useSA bool) (Client, error) {
```

Comment:
Style nit: I'd prefer an alternate `NewServiceAccountClient` or a varaidic option like `UsePodServiceAccount` over this boolean argument.